staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/algorithm.go

```diff
@@ -34,20 +39,26 @@ func Default(x interface{}, s *structuralschema.Structural) {
 			if prop.Default.Object == nil {
 				continue
 			}
-			if _, found := x[k]; !found {
+			if _, found := x[k]; !found || isNonNullableNull(x[k], &prop) {
 				x[k] = runtime.DeepCopyJSONValue(prop.Default.Object)
 			}
 		}
-		for k, v := range x {
+		for k := range x {
 			if prop, found := s.Properties[k]; found {
-				Default(v, &prop)
+				Default(x[k], &prop)
 			} else if s.AdditionalProperties != nil {
-				Default(v, s.AdditionalProperties.Structural)
+				if isNonNullableNull(x[k], s.AdditionalProperties.Structural) {
+					x[k] = runtime.DeepCopyJSONValue(s.AdditionalProperties.Structural.Default.Object)
```

Comment:
you don't check that there is a default.

Reply:
Does it matter? At that point, we know that `x[k]` is null, then that's going to reset it to null? That's a no-op.

Reply:
Yeah good point